lib/features/add_property/view_model/cubit/add_property_cubit.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:hatspace/features/add_property/view_model/cubit/add_property_state.dart';
+
+enum NavigatePage { forward, reverse }
+
+class AddPropertyCubit extends Cubit<AddPropertyState> {
+  AddPropertyCubit() : super(const AddPropertyInitial());
+
+  void navigatePage(NavigatePage navType, int totalPages) {
+    if (navType == NavigatePage.forward &&
+        state.pageViewNumber < totalPages - 1) {
+      emit(PageViewNavigationState(state.pageViewNumber + 1));
+    }
+    if (navType == NavigatePage.reverse && state.pageViewNumber > 0) {
+      emit(PageViewNavigationState(state.pageViewNumber - 1));
+      validateState(state.pageViewNumber);
+    }
+  }
+
+  List<bool> activePageList = [];
```

Comment:
this can be final.
And please move member variable definition to top of the class, right below constructor